File: Chu_de_1_2/app/src/main/java/com/example/chu_de_1_2/adapters/StudentAdapter.java
package com.example.chu_de_1_2.adapters;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.example.chu_de_1_2.databinding.LayoutItemStudentBinding;
import com.example.chu_de_1_2.entities.Student;

import java.util.ArrayList;

public class StudentAdapter extends RecyclerView.Adapter<StudentAdapter.StudentViewHolder> {
    ArrayList<Student> listStudent;
    UpdateCallback updateCallback;

    public StudentAdapter(ArrayList<Student> listStudent) {
        this.listStudent = listStudent;
    }

    public ArrayList<Student> getListStudent() {
        return listStudent;
    }

    public void setListStudent(ArrayList<Student> listStudent) {
        this.listStudent = listStudent;
    }

    public void setUpdateCallback(UpdateCallback updateCallback) {
        this.updateCallback = updateCallback;
    }

    @NonNull
    @Override
    public StudentViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        return new StudentViewHolder(LayoutItemStudentBinding.inflate(LayoutInflater.from(parent.getContext()), parent, false));
    }

    @Override
    public void onBindViewHolder(@NonNull StudentViewHolder holder, int position) {
        holder.bind(holder.getAdapterPosition());
    }

    @Override
    public int getItemCount() {
        return listStudent.size();
    }

    class StudentViewHolder extends RecyclerView.ViewHolder {
        private final LayoutItemStudentBinding binding;

        public StudentViewHolder(LayoutItemStudentBinding binding) {
            super(binding.getRoot());
            this.binding = binding;
        }

        void bind(int position) {
            binding.txtName.setText(listStudent.get(position).getName());
            binding.txtYear.setText(listStudent.get(position).getYear());
            binding.txtPhone.setText(listStudent.get(position).getPhone());
            binding.txtSpecialized.setText(listStudent.get(position).getSpecialized());
            binding.txtType.setText(listStudent.get(position).getType());
            binding.btnEdit.setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View view) {
                    updateCallback.selectedItem(position);
                }
            });

            binding.btnDelete.setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View view) {
                    listStudent.remove(listStudent.get(getAdapterPosition()));
                    notifyItemRemoved(getAdapterPosition());
                }
            });
        }
    }


    public interface UpdateCallback {
        void selectedItem(int position);
    }
}

File: Chu_de_1_2_Kotlin/app/src/main/java/com/example/chu_de_1_2_kotlin/view/StudentAdapter.kt
package com.example.chu_de_1_2_kotlin.view

import android.annotation.SuppressLint
import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView.Adapter
import androidx.recyclerview.widget.RecyclerView.ViewHolder
import com.example.chu_de_1_2_kotlin.databinding.ItemStudentBinding
import com.example.chu_de_1_2_kotlin.model.Student

class StudentAdapter : Adapter<StudentAdapter.StudentViewHolder>() {
    private var listStudent = arrayListOf<Student>()

    @SuppressLint("NotifyDataSetChanged")
    fun setData(list: ArrayList<Student>) {
        listStudent = list
        notifyDataSetChanged()
    }

    fun getData(): ArrayList<Student> {
        return listStudent
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): StudentViewHolder {
        val binding = ItemStudentBinding.inflate(LayoutInflater.from(parent.context), parent, false)
        return StudentViewHolder(binding)
    }

    override fun onBindViewHolder(holder: StudentViewHolder, position: Int) {
        holder.bind(position)
    }

    override fun getItemCount(): Int = listStudent.size

    inner class StudentViewHolder(private val binding: ItemStudentBinding) :
        ViewHolder(binding.root) {
        fun bind(position: Int) {
            binding.txtName.text = listStudent[position].name
            binding.txtYob.text = listStudent[position].yob
            binding.txtPhoneNumber.text = listStudent[position].phoneNumber
            binding.txtMajor.text = listStudent[position].major
            binding.txtType.text = listStudent[position].type
        }
    }
}